packages/mobile/src/account/LinkBankAccountScreen.tsx

```diff
@@ -22,92 +26,159 @@ import VerificationPending from 'src/icons/VerificationPending'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { linkBankAccountStepTwoEnabledSelector } from '../app/selectors'
+import { fetchFinclusiveKyc } from './actions'
 import openPlaid, { handleOnEvent } from './openPlaid'
 
-interface StepOneProps {
-  kycStatus: KycStatus | undefined
-}
-
 function LinkBankAccountScreen() {
-  // Log a cancel event on a "back" action (hardware back button, swipe, or normal navigate back)
   const navigation = useNavigation()
-  React.useEffect(() => {
+
+  // Log a cancel event on a "back" action (hardware back button, swipe, or normal navigate back)
+  useEffect(() => {
     const unsubscribe = navigation.addListener('beforeRemove', (e) => {
       ValoraAnalytics.track(CICOEvents.link_bank_account_cancel)
     })
     // Unsubscribe will be called on unmount
     return unsubscribe
   }, [])
 
-  const kycStatus = useSelector(kycStatusSelector)
-  const stepTwoEnabled = useSelector(linkBankAccountStepTwoEnabledSelector)
-
   return (
     <SafeAreaView style={styles.body}>
       <ScrollView
         contentContainerStyle={styles.contentContainer}
         showsVerticalScrollIndicator={false}
       >
-        <StepOne kycStatus={kycStatus} />
-        <StepTwo disabled={!stepTwoEnabled || kycStatus !== KycStatus.Approved} />
+        <StepOne />
+        <StepTwo />
       </ScrollView>
     </SafeAreaView>
   )
 }
 
-export function StepOne({ kycStatus }: StepOneProps) {
+interface StepOneUIStateParams {
+  kycStatus: KycStatus | undefined
+  finclusiveKycStatus: FinclusiveKycStatus
+  errorFromPersona: boolean
+  successFromPersona: boolean
+  isInPersonaFlow: boolean
+}
+
+enum StepOneUIState {
+  Begin = 'Begin',
+  Failure = 'Failure',
+  Pending = 'Pending',
+  Completed = 'Completed',
+  Spinner = 'Spinner',
+}
+export function stepOneUIState({
+  kycStatus,
+  finclusiveKycStatus,
+  errorFromPersona,
+  successFromPersona,
+  isInPersonaFlow,
+}: StepOneUIStateParams): StepOneUIState {
+  // The persona onSuccess / onError hooks fire slightly after the user is bounced back
+  // to the LinkedBankAccount Screen. We show a spinner at this time to avoid a more
+  // unpleasent screen flash as the kyc states change
+  if (isInPersonaFlow) {
+    return StepOneUIState.Spinner
+  }
+
+  if (finclusiveKycStatus === FinclusiveKycStatus.Accepted) {
+    return StepOneUIState.Completed
+  }
+
+  const userCompletedPersona: KycStatus[] = [
+    KycStatus.Completed,
+    KycStatus.Approved,
+    KycStatus.NeedsReview,
+  ]
+  const finclusiveNotCompleted = [FinclusiveKycStatus.Submitted, FinclusiveKycStatus.InReview]
+  if (
```

Comment:
If a user's Persona KYC status is `Approved`, but their Finclusive KYC status is `Rejected`, won't this return a `Pending` state here, since you're using an `or` operator here, when it should really return `Failure`?

Reply:
Good call out, definitely a mistake in this logic. I'll fix and write a test for it 👍

Reply:
Fixed